File: InterviewQuestions/ProductArray/productarray.c
#include <stdio.h>

int build_product_array(int input[], int size){
    int i,j;
    int result[size];
    if (size < 2){
        printf("Invalid input");
    }
    else{
        for(i=0; i<size; i++){
            int product = 1;
            for(j=0;j<size;j++){
                if(i != j){
                    product *= input[j];
                }
            }
            result[i] = product;
        }
        for (i=0;i<size;i++){
            printf("%d ", result[i]);
        }
    }
}

int main(){
    int input[] = {-1, 2, 3, 4, 5};
    int size = sizeof(input)/sizeof(int);
    build_product_array(input, size);
    return 0;
}

File: InterviewQuestions/ProductArray/productarray.py
# @param {List [int]} input
# @return {List [int]}
def build_product_array(input):
  # Your solution here

  results = [None for _ in range(len(input))]
  
  for i in range(len(input)): # iterate through the input
      product = 1 # reset the product
      for j in range(len(input)): # iterate through the input
          if i != j: # don't multiply by the same number
              product*=input[j] # multiply the product by the number
      results[i] = product # add the product to the results list

  return results # return the results

if __name__ == "__main__":
    input = [1, 2, 3, 4, 5]
    print(build_product_array(input))